iwyu_include_picker.cc

```diff
@@ -1329,18 +1329,50 @@ string IncludePicker::MaybeGetIncludeNameAsWritten(
   return value ? *value : "";
 }
 
-vector<string> IncludePicker::GetCandidateHeadersForSymbol(
-    const string& symbol) const {
-  CHECK_(has_called_finalize_added_include_lines_ && "Must finalize includes");
-  vector<MappedInclude> mapped_includes =
-      GetPublicValues(symbol_include_map_, symbol);
+vector<string> IncludePicker::BestQuotedIncludesForIncluder(
+    const vector<MappedInclude>& includes,
+    const string& including_filepath) const {
+  // Convert each MappedInclude to a quoted include, according to the
+  // following priorities:
+  // 1. If the file is already included, use whatever name it's already
+  //    included via.  This is better to use than ConvertToQuotedInclude
+  //    because it avoids trouble when the same file is accessible via
+  //    different include search-paths, or is accessed via a symlink.
+  // 2. If the quoted include in the MappedInclude object is an absolute path,
+  //    that's unlikely to be what's wanted.  Try to convert it to a relative
+  //    include via ConvertToQuotedInclude.
+  // 3. Othewise, use the quoted include present in the MappedInclude.
```

Comment:
Typo: Othewise

Other than that, this comment is awe-inspiring!

Reply:
Fixed.